# Réalisez une étude santé publique

In [1]:
## Question 1

In [2]:
import pandas as pd

In [3]:
import numpy as np

In [4]:
df1 = []

In [5]:
df1 = pd.read_csv("Q1.csv", usecols = ["Pays", "Valeur"])

In [6]:
df1

Pays   Valeur
0                               Afghanistan    30552
1                            Afrique du Sud    52776
2                                   Albanie     3173
3                                   Algérie    39208
4                                 Allemagne    82727
5                                    Angola    21472
6                        Antigua-et-Barbuda       90
7                           Arabie saoudite    28829
8                                 Argentine    41446
9                                   Arménie     2977
10                                Australie    23343
11                                 Autriche     8495
12                              Azerbaïdjan     9413
13                                  Bahamas      377
14                               Bangladesh   156595
15                                  Barbade      285
16                                  Bélarus     9357
17                                 Belgique    11104
18                                   Belize      332
19                                    Bénin    10323
20                                 Bermudes       65
21          Bolivie (État plurinational de)    10671
22                       Bosnie-Herzégovine     3829
23                                 Botswana     2021
24                                   Brésil   200362
25                        Brunéi Darussalam      418
26                                 Bulgarie     7223
27                             Burkina Faso    16935
28                               Cabo Verde      499
29                                 Cambodge    15135
..                                      ...      ...
145                                 Sénégal    14133
146                                  Serbie     9511
147                            Sierra Leone     6092
148                               Slovaquie     5450
149                                Slovénie     2072
150                                  Soudan    37964
151                               Sri Lanka    21273
152                                   Suède     9571
153                                  Suisse     8078
154                                Suriname      539
155                                Eswatini     1250
156                             Tadjikistan     8208
157                                   Tchad    12825
158                           Tchéquie (la)    10702
159                               Thaïlande    67011
160                             Timor-Leste     1133
161                                    Togo     6817
162                       Trinité-et-Tobago     1341
163                                 Tunisie    10997
164                            Turkménistan     5240
165                                 Turquie    74933
166                                 Ukraine    45239
167                                 Uruguay     3407
168                                 Vanuatu      253
169  Venezuela (République bolivarienne du)    30405
170                                Viet Nam    91680
171                                   Yémen    24407
172                                  Zambie    14539
173                                Zimbabwe    14150
174                                   Chine  1416667

[175 rows x 2 columns]

In [7]:
df1['Valeur'].sum() * 1000

8413993000

- Ce chiffre est supérieur à la population mondiale officielle. Il englobe deux fois la chine continentale.

In [8]:
df1_wo_china = df1[df1['Pays'] != "Chine"]

In [9]:
mondial_pop = df1_wo_china['Valeur'].sum() * 1000
print(mondial_pop)

6997326000


- C'est une valeur plus conforme aux chiffres officiels (Source: https://www.prb.org/wp-content/uploads/2014/02/2013-population-data-sheet_french.pdf)

In [10]:
temp = pd.read_csv("pop.csv")

In [11]:
temp.columns = ['xx','xx2','country_code','country','xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6']

In [12]:
pop = temp.pivot_table(
    index=["country_code","country","item_code","item","year"],
    columns = ["element"], values=["value"], aggfunc=sum)

In [13]:
pop.columns = ['total_population']

In [14]:
pop.reset_index()
pop.query('country == "France"')

total_population
country_code country item_code item       year                  
68           France  2501      Population 2013             64291

## Question 2

In [15]:
df2 = pd.read_csv("Q2.csv")

In [16]:
df2

Code Domaine              Domaine  Code Pays    Pays  Code Élément  \
0           FBS  Bilans Alimentaires         68  France          5511   
1           FBS  Bilans Alimentaires         68  France          5611   
2           FBS  Bilans Alimentaires         68  France          5072   
3           FBS  Bilans Alimentaires         68  France          5911   
4           FBS  Bilans Alimentaires         68  France          5301   
5           FBS  Bilans Alimentaires         68  France          5521   
6           FBS  Bilans Alimentaires         68  France          5527   
7           FBS  Bilans Alimentaires         68  France          5123   
8           FBS  Bilans Alimentaires         68  France          5131   
9           FBS  Bilans Alimentaires         68  France          5154   
10          FBS  Bilans Alimentaires         68  France          5142   
11          FBS  Bilans Alimentaires         68  France           645   
12          FBS  Bilans Alimentaires         68  France           664   
13          FBS  Bilans Alimentaires         68  France           674   
14          FBS  Bilans Alimentaires         68  France           684   

                                              Élément  Code Produit Produit  \
0                                          Production          2511     Blé   
1                             Importations - Quantité          2511     Blé   
2                                  Variation de stock          2511     Blé   
3                             Exportations - Quantité          2511     Blé   
4                            Disponibilité intérieure          2511     Blé   
5                               Aliments pour animaux          2511     Blé   
6                                            Semences          2511     Blé   
7                                              Pertes          2511     Blé   
8                                          Traitement          2511     Blé   
9                                 Autres Utilisations          2511     Blé   
10                                         Nourriture          2511     Blé   
11  Disponibilité alimentaire en quantité (kg/pers...          2511     Blé   
12     Disponibilité alimentaire (Kcal/personne/jour)          2511     Blé   
13  Disponibilité de protéines en quantité (g/pers...          2511     Blé   
14  Disponibilité de matière grasse en quantité (g...          2511     Blé   

    Code Année  Année               Unité    Valeur Symbole  \
0         2013   2013  Milliers de tonnes  38614.00       S   
1         2013   2013  Milliers de tonnes   2055.00       S   
2         2013   2013  Milliers de tonnes   1131.00       S   
3         2013   2013  Milliers de tonnes  21502.00       S   
4         2013   2013  Milliers de tonnes  20298.00       S   
5         2013   2013  Milliers de tonnes   7822.00       S   
6         2013   2013  Milliers de tonnes    748.00       S   
7         2013   2013  Milliers de tonnes    358.00       S   
8         2013   2013  Milliers de tonnes   1575.00       S   
9         2013   2013  Milliers de tonnes   2824.00       S   
10        2013   2013  Milliers de tonnes   6971.00       S   
11        2013   2013                  kg    108.43      Fc   
12        2013   2013  Kcal/personne/jour    811.00      Fc   
13        2013   2013     g/personne/jour     25.99      Fc   
14        2013   2013     g/personne/jour      4.20      Fc   

   Description du Symbole  
0   Données standardisées  
1   Données standardisées  
2   Données standardisées  
3   Données standardisées  
4   Données standardisées  
5   Données standardisées  
6   Données standardisées  
7   Données standardisées  
8   Données standardisées  
9   Données standardisées  
10  Données standardisées  
11        Donnée calculée  
12        Donnée calculée  
13        Donnée calculée  
14        Donnée calculée

- Disponibilité intérieure = Production + Importations - Exportations + Variation des stocks = Nourriture + Aliments pour animaux + Semences + Pertes + Traitement + Autres Utilisations

- Nourriture / population = Disponibilité alimentaire en quantité (kg par personne par an)

## Question 3

In [17]:
# Import des données des bilans alimentaires
veg = pd.read_csv("veg.csv")
ani = pd.read_csv("ani.csv")

# Ajout de la variable origin
ani["origin"] = "animal"
veg["origin"] = "vegetal"

# On regroupe veg et ani en un unique dataframe, via une union
temp = ani.append(veg)

# Suppression de ani et veg
del ani, veg

# On renomme les colonnes de temp
temp.columns = ["xx","xx2","country_code","country",'xx3','element'
    ,'item_code','item','xx4',"year","unit","value",'xx5','xx6'
    ,'origin']

# Transformation de temp en table pivot
data = temp.pivot_table(
    index=["country_code","country","item_code","item","year","origin"],
    columns = ["element"], values=["value"], aggfunc=sum)
# On renomme les colonnes (attention l'ordre peut changer selon vos données !)
data.columns = ['domestic_supply_quantity', 'export_quantity', 'fat_supply_quantity_gcapitaday',
    'feed', 'food', 'food_supply_kcalcapitaday', 
    'food_supply_quantity_kgcapitayr', 'import_quantity', 'waste', 'other_uses',
    'processing', 'production', 'protein_supply_quantity_gcapitaday', 
    'seed', 'stock_variation']
    
data = data.reset_index()
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  food_supply_kcalcapitaday  food_supply_quantity_kgcapitayr  \
0   93.0  ...                     1024.0                           130.60   
1  137.0  ...                        0.0                             0.00   
2   96.0  ...                        0.0                             0.03   
3    1.0  ...                        1.0                             0.12   
4    4.0  ...                        2.0                             0.37   

   import_quantity  waste  other_uses  processing  production  \
0            361.0   32.0         0.0        10.0       312.0   
1              9.0   15.0        26.0         7.0       189.0   
2             82.0    7.0         NaN         NaN        21.0   
3              0.0    0.0         NaN         NaN         1.0   
4              1.0    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  
0                               30.52  30.0           -118.0  
1                                0.00  14.0              0.0  
2                                0.01   0.0              NaN  
3                                0.02   0.0              0.0  
4                                0.09   0.0              NaN  

[5 rows x 21 columns]

In [18]:
data = data.merge(pop, on=['country'])
data = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["food_supply_kcalcapitaday"], how="all")
data = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["protein_supply_quantity_gcapitaday"], how="all")
data = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["food_supply_quantity_kgcapitayr"], how="all")

In [19]:
data['total_kcal'] = data['food_supply_kcalcapitaday'] * 365 * data['total_population'] * 1000

In [20]:
data['total_protein_kg'] = (data['protein_supply_quantity_gcapitaday'] / 1000) * 365 * data['total_population'] * 1000
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  waste  other_uses  processing  production  \
0   93.0  ...   32.0         0.0        10.0       312.0   
1  137.0  ...   15.0        26.0         7.0       189.0   
2   96.0  ...    7.0         NaN         NaN        21.0   
3    1.0  ...    0.0         NaN         NaN         1.0   
4    4.0  ...    0.0         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg  
0              2977  1.112684e+12       33163184.60  
1              2977  0.000000e+00              0.00  
2              2977  0.000000e+00          10866.05  
3              2977  1.086605e+09          21732.10  
4              2977  2.173210e+09          97794.45  

[5 rows x 24 columns]

## Question 4

In [21]:
data['kcal/kg'] = data['food_supply_kcalcapitaday'] * 365 / data['food_supply_quantity_kgcapitayr']
data.head()

country_code  country  item_code                 item  year   origin  \
0             1  Armenia       2511   Wheat and products  2013  vegetal   
1             1  Armenia       2513  Barley and products  2013  vegetal   
2             1  Armenia       2514   Maize and products  2013  vegetal   
3             1  Armenia       2515     Rye and products  2013  vegetal   
4             1  Armenia       2516                 Oats  2013  vegetal   

   domestic_supply_quantity  export_quantity  fat_supply_quantity_gcapitaday  \
0                     554.0              1.0                            3.60   
1                     198.0              0.0                            0.00   
2                     102.0              NaN                             NaN   
3                       1.0              NaN                            0.00   
4                       6.0              NaN                            0.03   

    feed  ...  other_uses  processing  production  \
0   93.0  ...         0.0        10.0       312.0   
1  137.0  ...        26.0         7.0       189.0   
2   96.0  ...         NaN         NaN        21.0   
3    1.0  ...         NaN         NaN         1.0   
4    4.0  ...         NaN         NaN         5.0   

   protein_supply_quantity_gcapitaday  seed  stock_variation  \
0                               30.52  30.0           -118.0   
1                                0.00  14.0              0.0   
2                                0.01   0.0              NaN   
3                                0.02   0.0              0.0   
4                                0.09   0.0              NaN   

   total_population    total_kcal  total_protein_kg      kcal/kg  
0              2977  1.112684e+12       33163184.60  2861.868300  
1              2977  0.000000e+00              0.00          NaN  
2              2977  0.000000e+00          10866.05     0.000000  
3              2977  1.086605e+09          21732.10  3041.666667  
4              2977  2.173210e+09          97794.45  1972.972973  

[5 rows x 25 columns]

In [22]:
temp = data[data['item'].str.contains("Eggs")]

In [23]:
# valeur nutritionnelle pour 100g
egg_ve = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
egg_ve['kcal/kg'].mean()/10

136.47554180345907

- ce chiffre est inférieur en moyenne dans le monde à la valeur de 147 kcal pour un oeuf de poule entier cru donné par Wikipedia (https://fr.wikipedia.org/wiki/Œuf_(aliment))

In [24]:
data['protein_percentage'] = ((data['protein_supply_quantity_gcapitaday'] * 1000 * 365) / (data['food'] * 1000000)) * 100

In [25]:
data.head(20)

country_code  country  item_code                          item  year  \
0              1  Armenia       2511            Wheat and products  2013   
1              1  Armenia       2513           Barley and products  2013   
2              1  Armenia       2514            Maize and products  2013   
3              1  Armenia       2515              Rye and products  2013   
4              1  Armenia       2516                          Oats  2013   
7              1  Armenia       2520                Cereals, Other  2013   
8              1  Armenia       2531         Potatoes and products  2013   
16             1  Armenia       2549    Pulses, Other and products  2013   
17             1  Armenia       2551             Nuts and products  2013   
19             1  Armenia       2556       Groundnuts (Shelled Eq)  2013   
21             1  Armenia       2558          Rape and Mustardseed  2013   
22             1  Armenia       2560         Coconuts - Incl Copra  2013   
23             1  Armenia       2561                   Sesame seed  2013   
24             1  Armenia       2563  Olives (including preserved)  2013   
28             1  Armenia       2573             Sunflowerseed Oil  2013   
37             1  Armenia       2586           Oilcrops Oil, Other  2013   
38             1  Armenia       2601         Tomatoes and products  2013   
39             1  Armenia       2602                        Onions  2013   
40             1  Armenia       2605             Vegetables, Other  2013   
41             1  Armenia       2611           Oranges, Mandarines  2013   

     origin  domestic_supply_quantity  export_quantity  \
0   vegetal                     554.0              1.0   
1   vegetal                     198.0              0.0   
2   vegetal                     102.0              NaN   
3   vegetal                       1.0              NaN   
4   vegetal                       6.0              NaN   
7   vegetal                      21.0              0.0   
8   vegetal                     672.0              9.0   
16  vegetal                       3.0              0.0   
17  vegetal                       6.0              1.0   
19  vegetal                       1.0              0.0   
21  vegetal                       0.0              NaN   
22  vegetal                       0.0              0.0   
23  vegetal                       0.0              NaN   
24  vegetal                       3.0              1.0   
28  vegetal                      30.0              0.0   
37  vegetal                       2.0              0.0   
38  vegetal                     273.0              3.0   
39  vegetal                      53.0              0.0   
40  vegetal                     764.0              6.0   
41  vegetal                      14.0              1.0   

    fat_supply_quantity_gcapitaday   feed  ...  processing  production  \
0                             3.60   93.0  ...        10.0       312.0   
1                             0.00  137.0  ...         7.0       189.0   
2                              NaN   96.0  ...         NaN        21.0   
3                             0.00    1.0  ...         NaN         1.0   
4                             0.03    4.0  ...         NaN         5.0   
7                             0.04    1.0  ...         NaN        17.0   
8                             0.12  225.0  ...        78.0       661.0   
16                             NaN    3.0  ...         NaN         0.0   
17                            1.76    NaN  ...         NaN         5.0   
19                            0.55    NaN  ...         NaN         NaN   
21                            0.00    0.0  ...         NaN         NaN   
22                            0.03    NaN  ...         NaN         NaN   
23                            0.01    NaN  ...         NaN         NaN   
24                            0.28    NaN  ...         NaN         NaN   
28                           24.38    NaN  ...         NaN         1.0   
37  

In [26]:
temp = data[data['item'].str.contains("Oats")]

In [27]:
temp = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["protein_percentage"], how="all")

In [28]:
temp = temp[temp['protein_supply_quantity_gcapitaday'] > 0]

In [29]:
temp['protein_percentage'].mean()

2.080560270459758

- La teneur en protéine de l'avoine est en moyenne de 2,1%

## Question 5

In [30]:
temp = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["kcal/kg"], aggfunc=np.mean).sort_values(by='kcal/kg', ascending=False).head(20)

kcal/kg
item_code item                     year origin              
2579      Sesameseed Oil           2013 vegetal  9559.523810
2582      Maize Germ Oil           2013 vegetal  8950.891229
2580      Olive Oil                2013 vegetal  8774.038462
2573      Sunflowerseed Oil        2013 vegetal  8769.634357
2571      Soyabean Oil             2013 vegetal  8739.687728
2577      Palm Oil                 2013 vegetal  8682.340531
2586      Oilcrops Oil, Other      2013 vegetal  8417.429666
2581      Ricebran Oil             2013 vegetal  8264.150943
2574      Rape and Mustard Oil     2013 vegetal  8238.956406
2578      Coconut Oil              2013 vegetal  7912.109869
2575      Cottonseed Oil           2013 vegetal  7708.151847
2576      Palmkernel Oil           2013 vegetal  7654.792329
2737      Fats, Animals, Raw       2013 animal   7562.536349
2572      Groundnut Oil            2013 vegetal  7208.090833
2781      Fish, Body Oil           2013 animal   7133.945055
2740      Butter, Ghee             2013 animal   6973.406757
2562      Palm kernels             2013 vegetal  5475.000000
2556      Groundnuts (Shelled Eq)  2013 vegetal  5466.240153
2561      Sesame seed              2013 vegetal  5144.359508
2633      Cocoa Beans and products 2013 vegetal  4342.347542

- Parmi les 20 aliments les plus caloriques, on peut citer toutes les huiles, les graisses animales, les cacahuètes et le beurre 

In [31]:
temp = data.replace([np.inf, -np.inf], np.nan).dropna(subset=["protein_percentage"], how="all")
temp.pivot_table(
    index=["item_code","item","year","origin"],
    values=["protein_percentage"], aggfunc=np.mean).sort_values(by='protein_percentage', ascending=False).head(20)

protein_percentage
item_code item                       year origin                     
2734      Poultry Meat               2013 animal            12.341237
2731      Bovine Meat                2013 animal            11.315274
2733      Pigmeat                    2013 animal            10.015137
2763      Pelagic Fish               2013 animal             9.496129
2549      Pulses, Other and products 2013 vegetal            8.985490
2764      Marine Fish, Other         2013 animal             8.773284
2511      Wheat and products         2013 vegetal            7.620493
2736      Offals, Edible             2013 animal             6.747166
2556      Groundnuts (Shelled Eq)    2013 vegetal            6.529279
2744      Eggs                       2013 animal             6.449286
2805      Rice (Milled Equivalent)   2013 vegetal            6.409195
2555      Soyabeans                  2013 vegetal            5.855895
2762      Demersal Fish              2013 animal             5.741086
2547      Peas                       2013 vegetal            5.626240
2546      Beans                      2013 vegetal            5.362527
2732      Mutton & Goat Meat         2013 animal             4.518095
2735      Meat, Other                2013 animal             4.448258
2765      Crustaceans                2013 animal             4.147485
2514      Maize and products         2013 vegetal            3.441944
2848      Milk - Excluding Butter    2013 animal             3.427677

- Parmi les 20 aliments les plus riches en protéines, on peut citer:
    les viandes & poissons, les lentilles, la farine, le soja, les noix

## Question 6

In [32]:
vegetals = data[data['origin'] == "vegetal"]

## Question 7

In [33]:
(vegetals['total_kcal'].sum()/(1819*365))/1000000000

7.754478344145135

In [34]:
(vegetals['total_kcal'].sum()/(2500*365))/1000000000

5.6421584432

- La disponibilité intérieure mondiale de produits végétaux permettrait de nourrir 7,75 milliards de personnes
(à raison d'un apport de 1819 Calories nécessaires par jour et par personne pour assurer une activité physique légère et maintenir un poids minimum acceptable.) source: http://www.fao.org/3/i3434f/i3434f02.pdf
- Et 5,6 milliards pour un apport moyen de 2500 Calories

In [35]:
(vegetals['total_kcal'].sum()/(2500*365)) / mondial_pop * 100

80.63306530523231

- Soit 80,6% de la population mondiale

In [36]:
(vegetals['total_protein_kg'].sum()/(0.04*365))/1000000000

8.601666658250002

- La disponibilité intérieure mondiale de protéines végétales est suffisante pour 8,6 milliards de personnes
(à raison d'un apport moyen de 40g de protéines par jour et par personne) source: http://www.fao.org/3/x0081f/X0081F0p.htm

In [51]:
(vegetals['total_protein_kg'].sum()/(0.04*365))/ mondial_pop * 100

122.92791072261036

- Soit 123 % de la population mondiale

## Question 8

In [37]:
temp = vegetals.copy()
temp = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
temp['total_food_kcal'] = (temp['food'] + temp['feed'] + temp['waste']) * 1000000 * temp['kcal/kg']
temp['total_food_kcal'].sum()/(2500*365)/1000000000

6.726453395149421

- En additionant, les quantités de nourriture d'origine végétale destinées à l'alimentation humaine et animale avec la nourriture gaspillée, on pourrait nourrir 6,7 milliards de personnes

In [38]:
(temp['total_food_kcal'].sum()/(2500*365)) / mondial_pop * 100

96.12891260389212

- Soit 96,1 % de la population mondiale

## Question 9

In [39]:
temp = data.copy()
temp = temp.replace([np.inf, -np.inf], np.nan).dropna(subset=["kcal/kg"], how="all")
temp['total_food_kcal'] = (temp['food']) * 1000000 * temp['kcal/kg']
temp['total_food_kcal'].sum()/(2500*365)/1000000000

7.087250976795807

In [40]:
(temp['total_food_kcal'].sum()/(2500*365)) / mondial_pop *100

101.285133446631

- Soit un peu plus de la totalité des êtres humains sur la planète !

## Question 10

In [41]:
df10 = pd.read_csv("Q10.csv", usecols = ["Produit", "Valeur"])

In [42]:
(df10['Valeur'] * 1000 / pop['total_population'].sum()) * 100

0    11.67829
Name: Valeur, dtype: float64

- Le pourcentage de la population mondiale en situation de sous-nutrition est de 11.7 %

## Question 11

In [43]:
# Liste des codes produit classés par la FAO dans la catégorie Céréales
cereals = [2511, 2513, 2514, 2515, 2516, 2517, 2518, 2520, 2805]

for item_code in cereals:
    data['is_cereal'] = np.where(data['item_code'] == item_code, True, False)

# Test if values have been updated as wanted
data.query('item_code == 2805')

country_code                             country  item_code  \
87                1                             Armenia       2805   
147               2                         Afghanistan       2805   
239               3                             Albania       2805   
332               4                             Algeria       2805   
412               7                              Angola       2805   
503               8                 Antigua and Barbuda       2805   
596               9                           Argentina       2805   
691              10                           Australia       2805   
786              11                             Austria       2805   
876              12                             Bahamas       2805   
968              14                            Barbados       2805   
1062             16                          Bangladesh       2805   
1143             17                             Bermuda       2805   
1236             19    Bolivia (Plurinational State of)       2805   
1330             20                            Botswana       2805   
1426             21                              Brazil       2805   
1515             23                              Belize       2805   
1593             25                     Solomon Islands       2805   
1688             26                   Brunei Darussalam       2805   
1781             27                            Bulgaria       2805   
1866             28                             Myanmar       2805   
1960             32                            Cameroon       2805   
2056             33                              Canada       2805   
2138             35                          Cabo Verde       2805   
2223             37            Central African Republic       2805   
2318             38                           Sri Lanka       2805   
2391             39                                Chad       2805   
2484             40                               Chile       2805   
2580             44                            Colombia       2805   
2669             46                               Congo       2805   
...             ...                                 ...        ...   
12989           211                         Switzerland       2805   
13047           213                        Turkmenistan       2805   
13140           214           China, Taiwan Province of       2805   
13236           215         United Republic of Tanzania       2805   
13331           216                            Thailand       2805   
13421           217                                Togo       2805   
13512           220                 Trinidad and Tobago       2805   
13606           221                                Oman       2805   
13698           222                             Tunisia       2805   
13791           223                              Turkey       2805   
13883           225                United Arab Emirates       2805   
13978           226                              Uganda       2805   
14073           229                      United Kingdom       2805   
14164           230                             Ukraine       2805   
14259           231            United States of America       2805   
14347           233                        Burkina Faso       2805   
14440           234                             Uruguay       2805   
14523           235                          Uzbekistan       2805   
14619           236  Venezuela (Bolivarian Republic of)       2805   
14700           237                            Viet Nam       2805   
14791           238                            Ethiopia       2805   
14867           244                               Samoa       2805   
14960           249                               Yemen       2805   
15053           251                              Zambia       2805   
15148           255                             Belgium       2805   
15241           2

In [44]:
data.query('item_code == 2543')

country_code                             country  item_code  \
520               9                           Argentina       2543   
710              11                             Austria       2543   
1349             21                              Brazil       2543   
1705             27                            Bulgaria       2543   
1979             33                              Canada       2543   
2241             38                           Sri Lanka       2543   
2408             40                               Chile       2543   
2689             48                          Costa Rica       2543   
3406             57                             Belarus       2543   
3969             67                             Finland       2543   
4064             68                              France       2543   
4596             79                             Germany       2543   
4688             80              Bosnia and Herzegovina       2543   
4947             84                              Greece       2543   
5580             96                China, Hong Kong SAR       2543   
5767             98                             Croatia       2543   
5952            100                               India       2543   
6048            101                           Indonesia       2543   
6143            102          Iran (Islamic Republic of)       2543   
6321            104                             Ireland       2543   
6507            106                               Italy       2543   
6789            109                             Jamaica       2543   
6884            110                               Japan       2543   
7408            117                   Republic of Korea       2543   
7592            119                              Latvia       2543   
7971            126                           Lithuania       2543   
8153            129                          Madagascar       2543   
8338            131                            Malaysia       2543   
8885            138                              Mexico       2543   
10252           162                              Norway       2543   
10347           165                            Pakistan       2543   
10441           166                              Panama       2543   
10535           167                             Czechia       2543   
10628           169                            Paraguay       2543   
10818           171                         Philippines       2543   
10910           173                              Poland       2543   
11325           183                             Romania       2543   
11930           194                        Saudi Arabia       2543   
12292           199                            Slovakia       2543   
12387           202                        South Africa       2543   
12482           203                               Spain       2543   
12913           211                         Switzerland       2543   
13065           214           China, Taiwan Province of       2543   
13255           216                            Thailand       2543   
13439           220                 Trinidad and Tobago       2543   
13716           223                              Turkey       2543   
13997           229                      United Kingdom       2543   
14183           231            United States of America       2543   
14365           234                             Uruguay       2543   
14543           236  Venezuela (Bolivarian Republic of)       2543   
15072           255                             Belgium       2543   
15526           351                               China       2543   

                    item  year   origin  domestic_supply_quantity  \
520    Sweeteners, Other  2013  vegetal                     279.0   
710    Sweeteners, Other  2013  vegetal                     294.0   
1349   Sweeteners, Other  2013  vegetal                     142.0   
1705   Sweeteners, O

In [45]:
temp = data[data['is_cereal'] == True]

temp['feed'].sum() / (temp['feed'].sum() + temp['food'].sum()) * 100

8.162745635371978

- La part de l'alimentation en céréales destinée aux animaux est de 8,2% de l'alimentation totale

## Question 14

- En thaïlande, 6,1 millions de personnes sont sous alimentées en 2013

In [46]:
pop.query('country == "Thailand"')

total_population
country_code country  item_code item       year                  
216          Thailand 2501      Population 2013             67011

In [47]:
(6100000/(67011000)) * 100

9.102983092328126

- Soit 9.1 % de la population de ce pays.

In [48]:
temp = data[data['item'].str.contains("Cassava")]

In [49]:
temp = temp.query('country == "Thailand"')

In [50]:
temp['export_quantity'] / temp['production'] * 100

13248    83.41273
dtype: float64

- La proportion de manioc exportée de Thaïlande est de 83.4 %